internal/client/reading.go

```diff
@@ -27,5 +48,29 @@ type Reading struct {
 // Decode does NOT allocate under any condition. Additionally, it panics if b
 // isn't at least 40 bytes long.
 func (r *Reading) Decode(b []byte) (ok bool) {
-	panic(common.ErrNotImplemented)
+	if len(b) < 40 {
```

Comment:
Why not `_ = b[39]`?

Reply:
don't have specific preference but comparing with len allows providing custom message and just index out of bounds